op-node/rollup/driver/state.go

```diff
@@ -358,7 +360,9 @@ func (s *Driver) eventLoop() {
 			// no step, justified L1 information does not do anything for L2 derivation or status
 		case newL1Finalized := <-s.l1FinalizedSig:
 			s.l1State.HandleNewL1FinalizedBlock(newL1Finalized)
-			s.derivation.Finalize(newL1Finalized)
+			ctx, cancel := context.WithTimeout(s.driverCtx, time.Second*5)
+			s.finalizer.Finalize(ctx, newL1Finalized)
+			cancel()
```

Comment:
I _think_ `Finalize` could return an error and then we catch and log an error here which would make me feel less nervous about this and fix my comment about the errors not bubbling up from the `tryFinalize` function. It's still a little bit confusing because when the errors do bubble up we still don't reset so not strongly opinionated on this but it feels better.

Reply:
It's a side-effect of tests expecting us to finalize immediately upon receiving the signal. To finalize, there are some API calls that can happen (for sanity checks of L1 data). Previously it would only partially check the finality conditions, and not actually finalize in some valid cases. But now it relies on a `ctx` to actually perform the finalization correctly. If it's missed due to timeout, the finalization signal will still have been applied, and it will finalize when it checks finality on L1 derivation block-end. So not critical, and fine to log, but indeed a bit whack. I'm going with this to preserve the existing test assumptions. We can change those assumptions in a separate change later if still desired.